File: 백준/Gold/1253. 좋다/좋다.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

n = int(input())
    
arr = list(map(int, input().split()))
    
arr.sort()
cnt = 0
for i in range(n):
    s, e = 0, n-1
    while s < e:
        if s == i:
            s += 1
            continue
        if e == i:
            e -= 1
            continue
                
        sum = arr[s] + arr[e]
        
        if (sum == arr[i]):
            cnt += 1
            break
        elif (sum < arr[i]):
            s += 1
        else:
            e -= 1

print(str(cnt))

File: 백준/Gold/1253. 좋다/좋다.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n;
    cin >> n;
    vector<long> v(n);
    
    for (int i = 0; i < n; i++) {
        cin >> v[i];
    }
    
    sort(v.begin(), v.end());
    
    int cnt = 0;
    
    for (int i = 0; i < n; i++) {
        int s = 0;
        int e = n-1;
        while(s < e) {
            if (s == i) {
                s++;
                continue;
            } else if (e == i) {
                e--;
                continue;
            }
            long long sum = v[s] + v[e];
            if (sum == v[i]) {
                cnt++;
                break;
            } else if (sum < v[i]) {
                s++;
            } else {
                e--;
            }
        }
    }
    
    cout << cnt;
    
    
}